expr/sort.go

```diff
@@ -26,47 +26,32 @@ func isUnset(val zng.TypedEncoding) bool {
 // The returned function uses the same return conventions as standard
 // routines such as bytes.Compare() and strings.Compare(), so it may
 // be used with packages such as heap and sort.
-// A record in which one of the comparison fields is not present is
-// considered to be smaller than a record in which the field is present.
-// The handling of records in which a comparison field is unset is
-// governed by the unsetMax parameter.  If this parameter is true,
-// a record with unset is considered larger than a record with any other
-// value, and vice versa.
-func NewSortFn(unsetMax bool, fields ...FieldExprResolver) SortFn {
+// The handling of records in which a comparison field is unset (or
+// not present) is governed by the nilsMax parameter.  If this parameter
```

Comment:
nilsMax vs nullsFirst.... can we unify the names, at least for nil vs null?

Reply:
Looks like "nil" is the preferred term, will update...

Reply:
I'd say "unset" is the preferred term since that's what's used in zng/docs/spec.md and throughout the code. Why change?

Reply:
Hmm, interesting you bring that up @nwt. At one time several months back there was a moment where we tried to decide on our pick for unset/nil/null and as I recall @mccanne made a benevolent dictator call to go with "nil", which we started doing in ZQL as well as how we presented records in Desktop. It's still true in ZQL today, though I see in Desktop we now just show a ⦻ character. And as you point out, the ZNG spec and its many nods to our Zeek roots is making unset seem viable yet again. I suppose it wouldn't be expensive to switch it up if we have a different consensus, so lemme bring this up at sync today.

Reply:
There are two distinct cases: "this field exists but has the unset value in this record" and "this field is not present in this record".  To me, "unset" sounds like just the first case and I was trying to be clear that this covers both cases.  I don't mind changing back to unset if there's consensus for that though.

Reply:
I got rid of TypeUnset in #201.  It's been bugging me a long time.  Value{} is nil and Value{typ,nil} is unset and there are methods on Value to check the two cases.  I also changed zbuf.Record to handle these two cases properly.

Reply:
After some discussion at our team sync today and then some 1-on-1 with @mccanne, new consensus is to embrace `null` here (so, `sort -nulls last n`). Once we're doing that, to be consistent, we'll also start using `null` in the ZQL filter contexts where we'd previously been saying `nil`. The wider ZQL filter `nil`/`null` topic will be handled separately, but hopefully we're now unblocked on this PR.